File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const findMin = nums => {
    let left = 0, right = nums.length - 1;
    let min = -1;
    while (left <= right) {
        let mid = left + Math.floor((right - left) / 2);
        if (nums[mid] <= nums[nums.length - 1]) {
            min = nums[mid];
            right = mid - 1;
        } else {
            left = mid + 1;
        }
    }
    return min;
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        minimum = float('inf')
        left, right = 0, len(nums) - 1
        while left <= right:
            mid = left + (right - left) // 2
            if nums[mid] > nums[right]:
                left = mid + 1
            else:
                minimum = min(minimum, nums[mid])
                right = mid - 1
        return minimum
        